### DISCUSSION TITLE
Improved Pull Request comments for Monorepos

### DISCUSSION THREAD

**Discussion by Stupidism** [vercel/community#83] [Idea]
# Background
My company has a large monorepo, where several squads are working on dozens of projects every day. I just tried to use vercel for one project. But the comment from vercel shows up in every PR. Even after I configured `Ignore Build Step`.

![image](https://user-images.githubusercontent.com/5801015/141953317-06fb58ff-ef17-4152-ac43-45e9faf01f9b.png)

# Feature Request
Don't comment on PR unless the build passes `Ignore Build Step`.

**Comment by Stupidism** [comment]
@leerob Hi Leerob. Any chance to get this feature?

**Reply by leerob** [reply]
Absolutely, this is something we are exploring! We use monorepos internally and agree we want to improve this output further. Stay tuned 😁

**Reply by becca** [reply]
@Stupidism we are working on a comment redesign that should drastically improve the experience for monorepos. We can flag you in early to provide feedback on this feature. What is your team name on Vercel? If you feel more comfortable you can email me becca at vercel.com so we can make this happen 🚀

**Reply by rodilo** [reply]
Hi @becca can you also enable it for my team?
Team name is `creadi`.

Thanks 🙏

**Reply by becca** [reply]
this is now live for everyone 🚀 
https://vercel.com/changelog/improved-formatting-for-pull-request-comments

**Reply by becca** [reply]
@rodilo please feel free to share any feedback or concerns with the updated design. this is the first update and hope to share out more in the near future

**Reply by Xuanwo** [reply]
> this is now live for everyone rocket https://vercel.com/changelog/improved-formatting-for-pull-request-comments

Hi, I think there is something wrong.

The feature requested by this discussion is `Don't comment on PR unless the build passes Ignore Build Step.` However, I can't recognize the feature from this [update](https://vercel.com/docs/concepts/git/vercel-for-github).

To make it clear, we do need PR comments, so `"silent": true` doesn't work for us. But we don't need PR comments if the build has been ignored.

![image](https://user-images.githubusercontent.com/5351546/163921067-af03ac03-b9e1-4d60-a401-5ceb8a9eb45f.png)

**Reply by becca** [reply]
@Xuanwo you're correct. this was the first step towards us cleaning up the comment and separating out more clearly which deployments are in progress/successful versus canceled or ignored. our next iteration on this is to disable the comments for ignored builds. i'll keep you all posted on progress 🙌

**Reply by Xuanwo** [reply]
@becca Glad to see this feature is in development, looking forward to it!

**Comment by Xuanwo** [comment]
There is the same feature request: We have `website` in our project and only want vercel to build when files changed in `website`. But vercel always comments on PR ever when the build is canceled.

Discussion happened here: https://github.com/datafuselabs/databend/pull/3233#issuecomment-986020569
